src/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -94,6 +105,11 @@ internal override void Validate()
                 countOfCredentialTypesSpecified++;
             }
 
+            if (Config.ClientAssertion != null)
```

Comment:
This sort of validation is a code smell. It looks like the users can only really specify: a secret, a cert or a ClientAssertion. But never a ClientCredentialWrapper (this is an internal object for carrying all these options). This is really confusing, because the wrapper is a very thin layer that does absolutely nothing. 

Consider refactoring to either: 

- remove ClientCredentialWrapper 
- move a lot more logic that is scattered around into ClientCredentialWrapper

Reply:
Lets discuss in our 1:1 tomorow